src/app/common/stores/applications/Distribution/Distribution.store.ts

```diff
@@ -57,6 +61,18 @@ class DistributionStore extends BaseStore {
     });
   }
 
+  @action.bound
+  async cumulativeClaimed() {
+    // eslint-disable-next-line no-async-promise-executor
+    return new Promise(async (resolve) => {
+      const contract = this.distributionContract;
+      const walletStore: WalletStore = this.getStore('Wallet');
+      const result = await contract.methods.cumulativeClaimed(this.userAddress).call();
+      this.claimedRewards = walletStore.fromWei(parseInt(String(result)).toString());
+      resolve(result);
```

Comment:
Potential issue here is when someone will want to `const myvar = await cumulativeClaimed()` call, in this case you should `resolve(this.claimedRewards)` - a final value

Reply:
done